core/src/core/pmmr/pmmr.rs

```diff
@@ -665,14 +668,27 @@ pub fn family_branch(pos: u64, last_pos: u64) -> Vec<(u64, u64)> {
 }
 
 /// Gets the position of the rightmost node (i.e. leaf) beneath the provided subtree root.
-pub fn bintree_rightmost(num: u64) -> u64 {
-	num - bintree_postorder_height(num)
+pub fn bintree_rightmost(pos: u64) -> u64 {
+	pos - bintree_postorder_height(pos)
 }
 
 /// Gets the position of the rightmost node (i.e. leaf) beneath the provided subtree root.
-pub fn bintree_leftmost(num: u64) -> u64 {
-	let height = bintree_postorder_height(num);
-	num + 2 - (2 << height)
+pub fn bintree_leftmost(pos: u64) -> u64 {
+	let height = bintree_postorder_height(pos);
+	pos + 2 - (2 << height)
+}
+
+/// Iterator over all leaf pos beneath the provided subtree root (including the root itself).
+pub fn bintree_leaf_pos_iter(pos: u64) -> impl Iterator<Item = u64> {
+	let leaf_start = max(1, bintree_leftmost(pos as u64));
+	let leaf_end = bintree_rightmost(pos as u64);
+	(leaf_start..=leaf_end).filter(|x| is_leaf(*x))
```

Comment:
It would be more efficient to iterate over leaf_indices, and map those into node_indices with the very cheap
map( \leaf_index -> 2 * leaf_index - count_ones(leaf_index) )

Reply:
Is there a way to iterate over the leaf indices of a subtree without reading them from an already-stored bitmap, (which is implementation-specific and not available in these helpers which I assume are supposed to be 'pure' MMR operations)?

Reply:
i don't see what a bitmap has to do with this routine which is only about indices.
i'm suggesting to convert both leaf_start and leaf_end into leaf indices (which is as expensive as an is_leaf test), using a node_to_leaf_index method, then construct a range from those, and map each element back to a node index.

Reply:
thanks, that makes it clearer, will implement

Reply:
the max(1, part is also redundant here.

PS: I just made a PR rewriting that MMR doc to stick with 0-based positions.

Reply:
added a `pmmr_leaf_to_insertion_index` function along with some tests and modified `bintree_leaf_pos_iter` to use it. (Note this introduces error handling that wasn't previously needed, so there's still an awkward unwrap in there to deal with)